### ISSUE TITLE
Words in `\d-code` block disappear

### ISSUE THREAD

**Issue by nyuichi** [gfngfn/SATySFi#165]
To display the following program,

```Lean
inductive perm
| trans aaaaaa aaaaa : Π {l₁ l₂ l₃ : list α}, perm l₁ l₂ → perm l₂ l₃ → perm l₁ l₃

def mem_perm aaaaaaa aaaaa {t : α} : Π {Γ₁ Γ₂}, Γ₁ ~ Γ₂ → t ∈' Γ₁ → t ∈' Γ₂
```

compiling the following code,

```
@require: stdjareport
@require: code

document (|
  title = {\SATySFi;};
  author = {Nanashi Taro};
|) '<
+p{
  \d-code(`
  inductive perm
  | trans aaaaaa aaaaa : Π {l₁ l₂ l₃ : list α}, perm l₁ l₂ → perm l₂ l₃ → perm l₁ l₃

  def mem_perm aaaaaaa aaaaa {t : α} : Π {Γ₁ Γ₂}, Γ₁ ~ Γ₂ → t ∈' Γ₁ → t ∈' Γ₂
  `);
}
>
```

one gets the following pdf, in which `def` in the last line in the code block disappears.

<img width="1200" alt="2019-02-26 20 22 16" src="https://user-images.githubusercontent.com/540144/53409432-3e3d5d80-3a04-11e9-9b37-1acd8e94e2be.png">

When I encountered this bug, I was using a modified version of the code package in which the font is set to `Menlo` (to display unicode characters). To reproduce this with the default font I needed to insert some words in the code for "padding" ('aaaa...' in the program above). In case you could not reproduce this in your environment with the code I put above, probably you would need to insert (or get rid of) some words/characters.

I think there is a bug in the line wrapping procedure and using unicode characters has nothing to do with this bug (I'll try to minimize the reproducing code to check this when I have time).

**Comment by na4zagin3** [comment]
`\d-code` changes fonts only for Latin script. That’s the cause of broken characters.
https://github.com/gfngfn/SATySFi/blob/2d02a4757e407e2afa6634df9707a5ff48270baa/lib-satysfi/dist/packages/code.satyh#L68

Simply setting lmmono does not solve the problem as it lacks small Greek letters. 
We may have to find a new fonts which has Greek and have the same metrics as lmmono10, or a different font which has larger character repertoire.

**Comment by nyuichi** [comment]
@na4zagin3 

No, I didn't mean that unicode characters not being displayed in the pdf is the bug.
In my own version of the code package, font is set to Menlo both for Latin and OtherScript, but I still encounter the bug.

**Comment by nyuichi** [comment]
I've succeeded in reproducing the same bug with unicode-free code:

```satysfi
@require: stdjareport
@require: code

document (|
  title = {\SATySFi;};
  author = {Nanashi Taro};
|) '<
+p{
  \d-code(`
  inductive perm
  | trans aaaaaa aaaaa : a {la la la : list a}, perm la la a perm la la a perm la la

  def mem_perm aaaaaaa aaaaa {t : a} : a {a  a }, a ~ a a t a' a a t a' a
  `);
}
>
```

Again, `def` in the last line disappeared.

<img width="1200" alt="2019-02-26 21 38 37" src="https://user-images.githubusercontent.com/540144/53413279-e9ebab00-3a0e-11e9-98c1-7bc153d393da.png">

**Comment by na4zagin3** [comment]
Ah, sorry. I didn’t mean the bug where some characters disappear, but the other one where non-ASCII character were broken. I have yet no idea about the first one.
